functions/savePendingContacts.ts

```diff
@@ -0,0 +1,127 @@
+        return isError ? rejectCallback(callbackPayload) : resolveCallback(callbackPayload);
+      };
+
+      return saveContactHandler(context, { ...event, payload }, callback);
+    });
+};
+
+export const handler: ServerlessFunctionSignature = TokenValidator(
+  async (context: Context<EnvVars>, event: any, callback: ServerlessCallback) => {
+    const response = responseWithCors();
+    const resolve = bindResolve(callback)(response);
+
+    const { SYNC_SERVICE_SID, SAVE_CONTACT_FN } = context;
+
+    try {
+      if (!SAVE_CONTACT_FN) throw new Error('SAVE_CONTACT_FN env var not provided.');
+
+      const saveContactFn = getSaveContactFn(SAVE_CONTACT_FN, context, event);
+      if (!saveContactFn) {
+        return resolve(error500(new Error('Could not find a saveContact function'))); // Should it be HTTP 404?
+      }
+
+      const sharedStateClient = context.getTwilioClient().sync.services(SYNC_SERVICE_SID);
+      const list = await sharedStateClient.syncLists('pending-contacts').fetch();
+      const pendingContacts = await list.syncListItems().list();
+
+      type SyncListItemInstance = typeof pendingContacts[0];
+
+      const successfulRetriesListItems: SyncListItemInstance[] = [];
+      const failedRetriesListItems: SyncListItemInstance[] = [];
+
+      await Promise.all(
+        pendingContacts.map(listItem => {
+          const { payload } = listItem.data;
+
+          return saveContactFn(payload)
+            .then(() => successfulRetriesListItems.push(listItem))
+            .catch(() => failedRetriesListItems.push(listItem));
+        }),
+      );
+
+      const incrementRetries = (listItem: SyncListItemInstance) => {
+        const updateOptions = {
+          data: {
+            ...listItem.data,
+            retries: (listItem.data.retries || 0) + 1,
+          },
+        };
+
+        return listItem.update(updateOptions);
+      };
+      const removeFromPendingContacts = (listItem: SyncListItemInstance) => listItem.remove();
+
+      // Apply changes to pending-contacts sync list
+      await Promise.all<SyncListItemInstance | boolean>([
+        ...failedRetriesListItems.map(incrementRetries),
+        ...successfulRetriesListItems.map(removeFromPendingContacts),
+      ]);
+
+      const result = {
+        savedContacts: successfulRetriesListItems.length,
+        remainingPendingContacts: failedRetriesListItems.length,
+      };
+
+      return resolve(success(result));
+    } catch (err) {
+      if (
+        err instanceof Error &&
+        err.message &&
+        err.message.includes(
+          `The requested resource /Services/${SYNC_SERVICE_SID}/Lists/pending-contacts was not found`,
+        )
+      ) {
+        // Function should not fail or alarm if the sync list 'pending-contacts' doesn't exist yet
+        return resolve(success('The sync list pending-contacts was not found'));
+      }
+      // eslint-disable-next-line no-console
+      console.error(err);
```

Comment:
Could this result in a double countedCloudfront error for the purposes of alarm triggers?

Reply:
I remember we had a discussion where we've spotted that `console.error()` were being counted as errors in Twilio. But I don't remember if those `console.error()` were from Twilio Functions or from somewhere else. Do you remember?

If that's the case, we'd need to change all Twilio Functions. They're all using `console.error()` at the catch block.

Reply:
I think that's what @GPaoloni was saying, we should remove the `console.error` or reduce the level in all our serverless functions because they double count, can you confirm Gian?

Reply:
I've changed it to **console.warn()** on this serverless function only. Later on, we might change it for all other serverless functions.